install/kubernetes/cilium/charts/agent/templates/svc.yaml

```
@@ -17,3 +17,43 @@ spec:
   selector:
     k8s-app: cilium
 {{- end }}
+{{- if .Values.global.hubble.ui.enabled }}
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: hubble-grpc
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: cilium
+spec:
+  type: ClusterIP
+  clusterIP: None
+  selector:
+    k8s-app: cilium
+  ports:
+  - targetPort: 50051
+    protocol: TCP
+    port: 50051
+{{- end }}
+{{- if and .Values.global.hubble.metrics.enabled (.Values.global.hubble.metrics.serviceMonitor.enabled) }}
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: hubble-metrics
+  namespace: {{ .Release.Namespace }}
+  labels:
+    k8s-app: cilium
+spec:
+  clusterIP: None
+  type: ClusterIP
+  ports:
+  - name: metrics
+    port: {{ regexReplaceAll ":([0-9]+)$" .Values.global.hubble.metricsServer "${1}" }}
+    protocol: TCP
+    targetPort: metrics
```

```suggestion
  - name: hubble-metrics
    port: {{ regexReplaceAll ":([0-9]+)$" .Values.global.hubble.metricsServer "${1}" }}
    protocol: TCP
    targetPort: hubble-metrics
```

In addition, the target port here also needs to be defined (which it currently) doesn't. 

You will need to replace https://github.com/cilium/cilium/blob/794f5b3dc2b5f7b4848b2461c5d11104527b2077/install/kubernetes/cilium/charts/agent/templates/daemonset.yaml#L163-L169

with something like the following: 

For example:
```yaml
{{- if or .Values.global.prometheus.enabled .Values.global.hubble.metricsServer }}
        ports:
{{- if .Values.global.prometheus.enabled }}
        - containerPort: {{ .Values.global.prometheus.port }}
          hostPort: {{ .Values.global.prometheus.port }}
          name: prometheus
          protocol: TCP
{{- end }}
{{- if .Values.global.hubble.metricsServer }}
        - containerPort: {{ regexReplaceAll ":([0-9]+)$" .Values.global.hubble.metricsServer "${1}" }}
          hostPort: {{ regexReplaceAll ":([0-9]+)$" .Values.global.hubble.metricsServer "${1}" }}
          name: hubble-metrics
          protocol: TCP
{{- end }}
{{- end }}
```